src/app/signers/data/useSigners.ts

```diff
@@ -13,19 +13,42 @@ export interface SignerInfo {
   pooled_stacker_count: number;
   solo_stacker_count: number;
 }
-export function useSuspensePoxSigners(cycleId: string) {
+
+const DEFAULT_LIST_LIMIT = 100;
+
+const POX_SIGNERS_QUERY_KEY = 'pox-signers';
+
+const fetchPoxSigners = async (
+  apiUrl: string,
+  cycleId: number,
+  pageParam: number,
+  options: { limit?: number; offset?: number }
+): Promise<ApiResponseWithResultsOffset<PoxSigner>> => {
+  const limit = options.limit || DEFAULT_LIST_LIMIT;
+  const offset = pageParam || 0;
+  const queryString = new URLSearchParams({
+    limit: limit.toString(),
+    offset: offset.toString(),
+  }).toString();
+  const response = await fetch(
+    `${apiUrl}/extended/v2/pox/cycles/${cycleId}/signers${queryString ? `?${queryString}` : ''}`
+  );
+  return response.json();
+};
+
+export function useSuspensePoxSigners(
+  cycleId: number,
+  options: { limit?: number; offset?: number } = {}
+) {
   const { url: activeNetworkUrl } = useGlobalContext().activeNetwork;
 
-  return useSuspenseQuery<
-    ApiResponseWithResultsOffset<SignerInfo>,
-    Error,
-    ApiResponseWithResultsOffset<SignerInfo>
-  >({
-    queryKey: ['signers', cycleId],
-    queryFn: () =>
-      fetch(`${activeNetworkUrl}/extended/v2/pox/cycles/${cycleId}/signers`).then(
-        res => res.json() as Promise<ApiResponseWithResultsOffset<SignerInfo>>
-      ),
+  return useSuspenseInfiniteQuery<ApiResponseWithResultsOffset<PoxSigner>>({
+    queryKey: [POX_SIGNERS_QUERY_KEY, cycleId],
```

Comment:
same question around adding options to query key

Reply:
See previous comment